PeopleAndAppleStockPrices/PeopleAndAppleStockPrices/Controllers/StockListViewController.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  StockListViewController.swift
+//  PeopleAndAppleStockPrices
+//
+//  Created by Sunni Tang on 9/4/19.
+//  Copyright © 2019 Sunni Tang. All rights reserved.
+//
+
+import UIKit
+
+class StockListViewController: UIViewController {
+
+    @IBOutlet weak var stockListTableView: UITableView!
+    
+    var allStocks = [Stock]() {
+        didSet {
+            stockListTableView.reloadData()
+        }
+    }
+    
+    var headerNames = [String]()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureTableView()
+        loadStockData()
+        addSectionNames()
+    }
+    
+    private func configureTableView() {
+        stockListTableView.dataSource = self
+        stockListTableView.delegate = self
+    }
+    
+    private func loadStockData() {
+        let stockData = DataAPIClient.getStockDataFromJSON()
+        // TODO: Refactor later to not force unwrap?
+        allStocks = Stock.getAllStocks(from: stockData)!
+    }
+    
+    // TODO: Refactor to use error codes instead of fatalerror
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        guard let selectedIndex = stockListTableView.indexPathForSelectedRow else { fatalError("No cell was selected") }
+        guard segue.identifier == "stockCellToDetailSegue" else { fatalError("Unidentified segue") }
+        guard let stockDetailsVC = segue.destination as? StockDetailViewController else { fatalError("No destination View Controller") }
+        
+        //TODO: Refactor two lines of code below as a func to be reused throughout code
+        let stocksForSection = allStocks.filter { $0.dateForHeader == headerNames[selectedIndex.section] }
+        let currentStock = stocksForSection[selectedIndex.row]
+        
+        stockDetailsVC.stock = currentStock
+    }
+    
+}
+
+extension StockListViewController: UITableViewDelegate {
+    
```

Comment:
Great job tackling the section names. I know it wasn't easy. Maybe we can try to refactor it to do more in the model at some point?